docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,157 @@
+- Find out whether your idea fits with the scope and aims of the project. It's up to you to make a strong case to convince the project's developers of the merits of this feature. Keep in mind that we want features that will be useful to the majority of our users and not just a small subset. If you're just targeting a minority of users, consider writing an add-on/plugin library.
+
+<!-- omit in toc -->
+#### How Do I Submit a Good Enhancement Suggestion?
+
+Enhancement suggestions are tracked as [GitHub issues](https://github.com/MewsSystems/FuncSharp/issues).
+
+- Use a **clear and descriptive title** for the issue to identify the suggestion.
+- Provide a **step-by-step description of the suggested enhancement** in as many details as possible.
+- **Describe the current behavior** and **explain which behavior you expected to see instead** and why. At this point you can also tell which alternatives do not work for you.
+- You may want to **include screenshots and animated GIFs** which help you demonstrate the steps or point out the part which the suggestion is related to. You can use [this tool](https://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and [this tool](https://github.com/colinkeenan/silentcast) or [this tool](https://github.com/GNOME/byzanz) on Linux. <!-- this should only be included if the project has a GUI -->
+- **Explain why this enhancement would be useful** to most FuncSharp users. You may also want to point out the other projects that solved it better and which could serve as inspiration.
+
+<!-- You might want to create an issue template for enhancement suggestions that can be used as a guide and that defines the structure of the information to be included. If you do so, reference it here in the description. -->
+
+### Your First Code Contribution
+
+#### Environment setup
+
+All You need is a .NET 6 SDK, an IDE of your choice and you're good to go! 🎉
+
+#### Choose an issue
+
+Browse through the current list of issues and select one that captures your interest. Reach out to the repository maintainers to get the issue assigned to you!
+
+#### Submit a pull request
+
+Spot a minor enhancement you could make? Or are you already addressing an existing issue? Initiate a pull request, providing a comprehensive yet concise overview. The maintainers will promptly review and consider your contributions!
+
+### Improving The Documentation
+<!-- TODO
+Updating, improving and correcting the documentation
+
+-->
+
+## Styleguides
+### Commit Messages
+
+We propose a recommended approach for composing and comprehending commits: 
+
+```
+"This commit will <YOUR_COMMIT_MESSAGE>"
+``` 
+serving as a clear template. For instance, a commit could be `"This commit will <Introduce new data type>"`, providing a concise insight into the purpose of the commit.
```

Comment:
I wouldn't have the `This commit will` prefix - You even don't have it in the screenshot 😄